src/test/java/peoplesoft/model/employment/EmploymentTest.java

```diff
@@ -0,0 +1,179 @@
+import peoplesoft.model.job.Job;
+import peoplesoft.model.job.Money;
+import peoplesoft.model.job.Rate;
+import peoplesoft.model.util.ID;
+
+public class EmploymentTest {
+    private static final Job EATING = new Job(new ID(1043), "Eating",
+        new Rate(new Money(5.5), Duration.ofHours(1)), Duration.ofDays(1), false);
+    private static final Job RUNNING = new Job(new ID(3175), "Running",
+        new Rate(new Money(6), Duration.ofHours(4)), Duration.ofHours(8), true);
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Employment(null));
+    }
+
+    @Test
+    public void associate_null_throwsNullPointerException() {
+        // null job
+        assertThrows(NullPointerException.class, () -> new Employment().associate(null, ALICE));
+
+        // null person
+        assertThrows(NullPointerException.class, () -> new Employment().associate(EATING, null));
+    }
+
+    @Test
+    public void associate_withoutDuplicates_success() {
+        Employment employment = new Employment();
+
+        // Associate one person to a job
+        employment.associate(EATING, ALICE);
+        assertTrue(employment.getAllJobs().containsKey(EATING.getJobId()));
+        assertTrue(employment.getAllJobs().get(EATING.getJobId()).contains(ALICE.getPersonId()));
+
+        // Associate another person to the same job
+        employment.associate(EATING, BOB);
+        assertTrue(employment.getAllJobs().get(EATING.getJobId()).contains(BOB.getPersonId()));
+
+        // Associate an associated person to another job
+        employment.associate(RUNNING, ALICE);
+        assertTrue(employment.getAllJobs().containsKey(RUNNING.getJobId()));
+        assertTrue(employment.getAllJobs().get(RUNNING.getJobId()).contains(ALICE.getPersonId()));
+
+        // No association when not associated
+        assertFalse(employment.getAllJobs().get(RUNNING.getJobId()).contains(BOB.getPersonId()));
+    }
+
+    @Test
+    public void associate_duplicateEmployment_throwsDuplicateEmploymentException() {
+        Employment employment = new Employment();
+        employment.associate(EATING, ALICE);
+        assertThrows(DuplicateEmploymentException.class, () -> employment.associate(EATING, ALICE));
+    }
+
+    @Test
+    public void disassociate_null_throwsNullPointerException() {
+        // null job
+        assertThrows(NullPointerException.class, () -> new Employment().disassociate(null, ALICE));
+
+        // null person
+        assertThrows(NullPointerException.class, () -> new Employment().disassociate(EATING, null));
+    }
+
+    @Test
+    public void disassociate_employmentExists_success() {
+        Employment employment = new Employment();
+        employment.associate(EATING, ALICE);
+        employment.associate(EATING, BOB);
+        employment.associate(RUNNING, ALICE);
+
+        // Disassociate two jobs to same person
```

Comment:
huh

Reply:
what does "disassociate two jobs to the same person" mean, and why is there only one `disassociate()` here